File: model_obj/dimensions/dimMillimeter.m
classdef dimMillimeter < dimLinear
    %dimMillimeter Dimension in millimeters
    %   based off of the double class, this class implements required
    %   conversion methods.
     
   
    methods
        function obj = dimMillimeter(vargin)
            obj = obj@dimLinear(vargin);
        end
        
        function new = toMillimeter(obj)            
            %TOMILLIMETER Convert this dimension to mm
                        
            new = obj;
        end
        
        function new = toInch(obj) 
            %TOINCHES Convert this dimension to inches
            
           new = dimInch(obj/25.4); 
        end
    end
end

File: model_obj/dimensions/DimMillimeter.py

from DimLinear import DimLinear


class DimMillimeter(DimLinear):
    def __new__(cls, value):
        if type(value).__bases__[0] is DimLinear:
            value = value.to_mm()
        return DimLinear.__new__(cls, value)

    def to_mm(self):
        return self

    def to_inch(self):
        return self / 25.4

    def _to_dimes(self):

        return float(self)

    def _from_dimes(var):

        x = var
        return DimMillimeter(x)